invoiceregistry_test.go

```diff
@@ -172,7 +220,14 @@ func TestInvoiceExpiry(t *testing.T) {
 func TestAutoSettle(t *testing.T) {
 	defer test.Timeout()()
 
-	c := newRegistryTestContext(t, true)
+	c := newRegistryTestContext(t,
+		optAutoSettle(true),
+		optTestAmt(1000),
+
+		// Immediate timeout.
+		optAcceptTimeout(0),
+		optHtlcHoldDuration(0),
```

Comment:
Aren't these zero timeouts potentially causing race conditions?

Reply:
I've taken them out since they aren't super relevant to this test, but I don't believe so.

Auto-settle happens here, in `process()`:
https://github.com/bottlepay/lnmux/blob/master/invoiceregistry.go#L813

`process()` is called from `invoiceEventLoop` when an HTLC comes in here:
https://github.com/bottlepay/lnmux/blob/master/invoiceregistry.go#L297

This happens inside the same goroutine and in the same loop that handles the timeouts and the HTLC accepts here:
https://github.com/bottlepay/lnmux/blob/master/invoiceregistry.go#L277

I believe that means that the HTLCs get settled from lnmux's view before either of the timers would expire, even in an MPP situation where it's an HTLC that completes the set.

I did run the test (without any others at the same time) in this form with the `-race` flag and didn't see it trigger a race condition. However, after I changed the code back to initialize this test context with only autosettle, and ran the entire test suite with the `-race` flag, I noticed there were still races in `TestCancel` and `TestAutoSettle`. I've created #64 to address it.

Reply:
Indeed, I think you are right. The timeout channels don't even get to be evaluated before the settle happens.